src/app/cluster/cluster-details/alertmanager-config/alertmanager-config-dialog/component.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Inject, OnDestroy, OnInit} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {MLAService} from '@core/services/mla';
+import {NotificationService} from '@core/services/notification';
+import {AlertmanagerConfig} from '@shared/entity/mla';
+import * as _ from 'lodash';
+import {Subject} from 'rxjs';
+import {take} from 'rxjs/operators';
+
+export interface AlertmanagerConfigDialogData {
+  title: string;
+  projectId: string;
+  clusterId: string;
+  mode: Mode;
+  confirmLabel: string;
+
+  // Alertmanager Config has to be specified only if dialog is used in the edit mode.
+  alertmanagerConfig?: AlertmanagerConfig;
+}
+
+export enum Mode {
+  Edit = 'edit',
+}
+
+@Component({
+  selector: 'km-alertmanager-config-dialog',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class AlertmanagerConfigDialog implements OnInit, OnDestroy {
+  spec = '';
+  private readonly _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _matDialogRef: MatDialogRef<AlertmanagerConfigDialog>,
+    private readonly _mlaService: MLAService,
+    private readonly _notificationService: NotificationService,
+    @Inject(MAT_DIALOG_DATA) public data: AlertmanagerConfigDialogData
+  ) {}
+
+  ngOnInit(): void {
+    this._initProviderConfigEditor();
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  isValid(): boolean {
+    return !_.isEmpty(this.spec);
+  }
+
+  save(): void {
+    const alertmanagerConfig: AlertmanagerConfig = {
+      spec: {
+        config: this._getSpec(),
+      },
+    };
+
+    switch (this.data.mode) {
+      case Mode.Edit:
+        return this._edit(alertmanagerConfig);
+    }
+  }
+
+  private _initProviderConfigEditor(): void {
+    if (this.data.mode === Mode.Edit) {
+      const spec = this.data.alertmanagerConfig.spec.config;
+      if (!_.isEmpty(spec)) {
+        this.spec = atob(spec);
+      }
+    }
+  }
+
+  private _getSpec(): any {
```

Comment:
can we avoid using `any`? This actually should not even pass our linting rules.

Reply:
Good catch, was a missed leftover from work.